### ISSUE TITLE
Could you explain the design concept of @ToolMemoryId? I'm a bit confused about it.

### ISSUE THREAD

**Issue by Fj-ivy** [langchain4j/langchain4j#1459]
Why is the @ToolMemoryId annotation needed, when in the source code I only see that @MemoryId is set to the field annotated with @ToolMemoryId, and it is not actually used in ToolExecutor#execute? Could you explain the philosophy behind this design?

Thank you！

**Comment by langchain4j** [comment]
Hi @Fj-ivy I am not sure I understand your question.

`@ToolMemoryId` annotation is used in `DefaultToolExecutor` and `ToolSpecifications`.

Or is the question why we need a new (`@ToolMemoryId`) annotation if we could re-use the same `@MemoryId` anotation like so?

```java
interface Assistant {

    String chat(@MemoryId int memoryId, @UserMessage String userMessage);
}

class Tools {

    @Tool
    void addTask(@MemoryId int memoryId, Task task) {
        ...
    }
}
```

**Comment by Fj-ivy** [comment]
> Hi @Fj-ivy I am not sure I understand your question.
> 
> `@ToolMemoryId` annotation is used in `DefaultToolExecutor` and `ToolSpecifications`.
> 
> Or is the question why we need a new (`@ToolMemoryId`) annotation if we could re-use the same `@MemoryId` anotation like so?
> 
> ```java
> interface Assistant {
> 
>     String chat(@MemoryId int memoryId, @UserMessage String userMessage);
> }
> 
> class Tools {
> 
>     @Tool
>     void addTask(@MemoryId int memoryId, Task task) {
>         ...
>     }
> }
> ```

Yeah,  Essentially,  @ToolMemoryId and @MemoryId annotation are equal.  If you use two annotations, it might add to the learning burden。

**Comment by langchain4j** [comment]
@Fj-ivy I agree, introducing the new annotation was unnecessary, as an existing one could be re-used.
I think we could fix this by changing the logic to allow both `@ToolMemoryId` and `@MemoryId` to be used on `@Tool` method parameters and deprecate `@ToolMemoryId`, WDYT?
Would you be up to create a PR?

**Comment by Fj-ivy** [comment]
> @Fj-ivy I agree, introducing the new annotation was unnecessary, as an existing one could be re-used. I think we could fix this by changing the logic to allow both `@ToolMemoryId` and `@MemoryId` to be used on `@Tool` method parameters and deprecate `@ToolMemoryId`, WDYT? Would you be up to create a PR?

ok，I will create a PR！

**Comment by langchain4j** [comment]
Hi @Fj-ivy, do you still plan to open a PR? Thanks!
